src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3174,6 +3192,10 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
             MethodSymbol method = null)
         {
             Debug.Assert(!arguments.IsDefault);
+            bool methodHasNotNullIfNotNull = !IsAnalyzingAttribute &&
+                (method?.ReturnNotNullIfParameterNotNull.IsEmpty() == false ||
+                (!parameters.IsDefault && parameters.Any(p => !p.NotNullIfParameterNotNull.IsEmpty())));
+            var notNullParametersBuilder = methodHasNotNullIfNotNull ? ArrayBuilder<ParameterSymbol>.GetInstance() : null;
```

Comment:
FWIW, getting a builder is pooled, so if you're using `null` just to avoid allocations or overhead, getting an ArrayBuilder instance is incredibly cheap.

Reply:
Makes sense. My purpose wasn't to save on allocation but to not collect un-necessary information about parameter nullability when it is not relevant.

---
In reply to: [303558343](https://github.com/dotnet/roslyn/pull/37201#discussion_r303558343) [](ancestors = 303558343)